django_emarsys/event.py

```diff
@@ -0,0 +1,263 @@
+        return num_new_events, num_updated_ids, num_deleted_ids, \
+            unknown_event_names, unsynced_event_names
+
+    for local_event in Event.objects.all():
+        emarsys_event_id = emarsys_events.get(local_event.name)
+        if emarsys_event_id:
+            if local_event.emarsys_id != emarsys_event_id:
+                num_updated_ids += 1
+
+            local_event.emarsys_id = emarsys_event_id
+            del emarsys_events[local_event.name]
+            local_event.save()
+        else:
+            local_event.delete()
+            num_deleted_ids += 1
+
+    for emarsys_event_name, emarsys_event_id in emarsys_events.items():
+        Event.objects.create(emarsys_id=emarsys_event_id,
+                             name=emarsys_event_name)
+        num_new_events += 1
+
+    all_synced_event_names = set(Event.objects.all()
+                                 .values_list('name', flat=True))
+    known_event_names = set(settings.EMARSYS_EVENTS.keys())
+    unknown_event_names = list(all_synced_event_names - known_event_names)
+    unsynced_event_names = list(known_event_names - all_synced_event_names)
+
+    if unknown_event_names:
+        log.warning("these Emarsys event names are not "
+                    "configured in settings.EMARSYS_EVENTS: {}"
+                    .format(', '.join(
+                        '"{}"'.format(x.replace('"', '\\"'))
+                            for x in unknown_event_names)))
+
+    if unsynced_event_names:
+        log.warning("these event names in settings.EMARSYS_EVENTS are not "
+                    "known by Emarsys: {}"
+                    .format(', '.join(
+                        '"{}"'.format(x.replace('"', '\\"'))
+                            for x in unsynced_event_names)))
+
+    return num_new_events, num_updated_ids, num_deleted_ids, \
+        unknown_event_names, unsynced_event_names
+
+
+def trigger_automatic_event(event_name, recipient_email, data=None,
+                            create_user_if_needed=True,
+                            contact_data_provider=None):
+    return _trigger_event(event_name=event_name,
+                          recipient_email=recipient_email,
+                          data=data,
+                          create_user_if_needed=create_user_if_needed,
+                          contact_data_provider=contact_data_provider,
+                          source=EventInstance.SOURCE_AUTOMATIC)
+
+
+def trigger_manual_event(event_name, recipient_email, data=None,
+                         create_user_if_needed=True,
+                         contact_data_provider=None):
+    return _trigger_event(event_name=event_name,
+                          recipient_email=recipient_email,
+                          data=data,
+                          create_user_if_needed=create_user_if_needed,
+                          contact_data_provider=contact_data_provider,
+                          source=EventInstance.SOURCE_MANUAL)
```

Comment:
Yeah, I'll cite:
> [...] makes the choices easy to reference (e.g, Student.SOPHOMORE will work anywhere that the Student model has been imported)

That's a contradictio in adiectio in my opinion, because "anywhere the model has been imported" is, like, the opposite!, like, of "easy to reference" ^^

So the actual business constants should be defined outside the model. How the model uses these constants for its task is a debatable issue, and also one that is now irrelevant to this PR, since we have the *bool solution*.